back/controller/controller.go

```diff
@@ -76,3 +90,77 @@ func (c *Controller) Todos(w http.ResponseWriter, r *http.Request) {
 	}
 
 }
+
+func (c *Controller) Workers(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var done = myStruct{&sync.Mutex{}, 0}
+
+	type_str, items, max_items_per_worker, err := c.useCase.WorkerFlags(r)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprint(w, err.Error())
+		return
+	}
+	allPokemonsSlice, err := c.entity.GetAllPokemonsFromCSV()
+
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprint(w, err.Error())
+	}
+
+	values := make(chan models.Pokemon)
+	shutdown := make(chan struct{})
+	poolSize := len(allPokemonsSlice) / 3
+	steps := len(allPokemonsSlice) / poolSize
+	var wg sync.WaitGroup
+	wg.Add(poolSize)
+	rand.Seed(time.Now().UnixNano())
+	for i := 0; i < poolSize; i++ {
+		go func(ii int) {
+			rand.Seed(time.Now().UnixNano())
+			starting_index := steps * ii
+			ending_index := (steps + steps*ii) - 1
+			r := rand.Intn(ending_index-starting_index+1) + starting_index
+			items_of_worker := 0
+			for {
+				if items_of_worker >= max_items_per_worker {
+
+					done.mutex.Lock()
+					done.number += 1
+					done.mutex.Unlock()
+					wg.Done()
+					return
+				}
+				select {
+
+				case values <- allPokemonsSlice[r]:
+					items_of_worker++
+
+				case <-shutdown:
+					wg.Done()
+					return
+				}
+
+			}
+		}(i)
+	}
+	validPokemons := make([]models.Pokemon, 0)
+	numberOfValidPokemons := 0
+	for {
+		poke := <-values
+		if poke.Id%2 == 0 && type_str != "odd" {
+			validPokemons = append(validPokemons, poke)
+			numberOfValidPokemons++
+		} else if poke.Id%2 != 0 && type_str != "even" {
```

Comment:
```suggestion
		if poke.Id%2 == 0 && type_str == "even" {
			validPokemons = append(validPokemons, poke)
			numberOfValidPokemons++
		} else if poke.Id%2 != 0 && type_str == "odd" {
```

What do you think about changing the validations to these?